app/mailers/application_mailer.rb

```diff
@@ -22,6 +22,28 @@ class ApplicationMailer < ActionMailer::Base
   helper :application # gives access to all helpers defined within `application_helper`.
 
 
+  # if there is a problem communicating with the MailGun REST server, log the problem
+  def self.deliver_mail(mail)
+
+    begin
+      super
+
+    rescue => e
+
+      # This would be a great place to use the ExceptionNotification gem so we
+      # can also do other notifications in case we cannot communicate at all with MailGun.
+
+      # LOG the error
+      error_prefix = '>>> MAILGUN ERROR!  '
+      logger.debug( error_prefix + "Could not send email via mailgun at #{Time.now}" )
+      logger.debug{ error_prefix + " Error received from Mailgun: #{e}" }
+
+      raise e
```

Comment:
If the exception is due to a problem on the mailgun side (not available, for instance) re-raising is likely to cause a "red screen of death" to be seen by the user (unless we have a catch-all handler further up the stack).

To be discussed, but this is what I am doing currently in my HIPS spike, when an exception occurs with creating an order on HIPS:

```
rescue => e

  ActivityLogger.open('log/HIPS.log', 'HIPS_API', 'create order', false) do |log|
    log.record('error', 'Cannot create payment or HIPS order')
    log.record('error', "Payment ID: #{@payment&.id}")
    log.record('error', "HIPS ID: #{@hips_id if @hips_id}")
    log.record('error', "Exception class: #{e.class}")
    log.record('error', "Exception message: #{e.message}")
  end

  helpers.flash_message(:alert,
      'Something went wrong - please contact system administrator')

  redirect_to(request.referer.present? ? :back : root_path)
end
```